### ISSUE TITLE
Tensorflow Lite AllocateTensors() fails with custom model

### ISSUE THREAD

**Issue by phildue** [tensorflow/tensorflow#19380]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
binary
- **TensorFlow version (use command below)**:
- **Python version**: 
('v1.8.0-0-g93bc2e2072', '1.8.0')
- **Bazel version (if compiling from source)**:
```
Build label: 0.13.0
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Mon Oct 18 21:33:40 +50297 (1525078013620)
Build timestamp: 1525078013620
Build timestamp as int: 1525078013620

```
- **GCC/Compiler version (if compiling from source)**:
gcc version 5.5.0 20171010 (Ubuntu 5.5.0-12ubuntu1~16.04) 
- **CUDA/cuDNN version**:
None
- **GPU model and memory**:
None
- **Exact command to reproduce**:

### Python-Code:
```
from keras import Input, Model
from keras.layers import Conv2D
import keras.backend as K
import tensorflow as tf
from tensorflow.python.framework import graph_util
# manually put back imported modules
import tempfile
import subprocess

tf.contrib.lite.tempfile = tempfile
tf.contrib.lite.subprocess = subprocess

K.set_learning_phase(0)
input = Input((416, 416, 3),name='Placeholder')
conv1 = Conv2D(16, kernel_size=(3, 3), strides=(1, 1), padding='same', use_bias=False)(input)
model = Model(input, conv1)
netin = [K.placeholder(name="Input", dtype=tf.float32, shape=(416,416,3))]
netout = [K.identity(model.outputs[0],"Prediction")]
sess = K.get_session()

constant_graph = graph_util.convert_variables_to_constants(sess, sess.graph_def, ["Prediction"])

tflite_model = tf.contrib.lite.toco_convert(constant_graph, netin, netout)
open('model' + '.tflite', "wb").write(tflite_model)

```

### C-Code:

```
//based on tensorflow/contrib/lite/examples/minimal/minimal.cc

#include "tensorflow/contrib/lite/model.h"
#include "tensorflow/contrib/lite/interpreter.h"
#include "tensorflow/contrib/lite/kernels/register.h"
#include "tensorflow/contrib/lite/kernels/kernel_util.h"
#include <cstdio>
#include <iostream>
#include "exp.h"
#include "opencv2/highgui.hpp"
#include "opencv2/core.hpp"
// Usage: minimal <tflite model>

using namespace tflite;

#define TFLITE_MINIMAL_CHECK(x) \
  if(!(x)) {                                                    \
    fprintf(stderr, "Error at %s:%d\n",  __FILE__, __LINE__); \
    exit(1); \
  }


int main(int argc, char *argv[]) {
  if(argc != 3) {
    fprintf(stderr, "Usage: <model> <image>\n");
    return 1;
  }else{
      std::cout << "Reading model from: " << argv[1] << std::endl;
      std::cout << "Reading image from: " << argv[2] << std::endl;
  }
  const char* filename = argv[1];
    const char* imagefile = argv[2];

  // Load model
  std::unique_ptr<tflite::FlatBufferModel> model
      = tflite::FlatBufferModel::BuildFromFile(filename);
  TFLITE_MINIMAL_CHECK(model != nullptr);



  // Build the interpreter
  tflite::ops::builtin::BuiltinOpResolver resolver;
  resolver.AddCustom("Exp", Register_EXP());

  InterpreterBuilder builder(*model.get(), resolver);
  std::unique_ptr<Interpreter> interpreter;
  builder(&interpreter);
  TFLITE_MINIMAL_CHECK(interpreter != nullptr);
    cv::Mat cvimg = cv::imread(imagefile);
    cv::imshow("Input",cvimg);
    cv::waitKey(1);

  // Allocate tensor buffers.
    TFLITE_MINIMAL_CHECK(interpreter->AllocateTensors() == kTfLiteOk);
    // Fill input buffers

    int input = interpreter->inputs()[0];
    memcpy(interpreter->typed_input_tensor<float>(input), cvimg.data, cvimg.total() * cvimg.elemSize());

  // Run inference
    TFLITE_MINIMAL_CHECK(interpreter->Invoke() == kTfLiteOk);

  // Read output buffers
  // TODO(user): Insert getting data out code.
    //int output = interpreter->outputs()[0];
    std::cout << interpreter->typed_output_tensor<float>(0) << std::endl;
  return 0;
}
```
### Describe the problem

I convert a keras model (very simplified to reproduce error) to tflite. I verified the graph using tensorboard and it seems fine. However, when I want to run it in C++ the following error occurs during the call of interpreter->AllocateTensors():

```
tensorflow/contrib/lite/kernels/conv.cc:189 input->dims->size != 4 (0 != 4)
```
This does not happen when loading a model from [here](https://www.tensorflow.org/mobile/tflite/demo_android). What am I doing wrong?

**Comment by phildue** [comment]
Short update. This seems to be a keras -> tensorflow problem. When I define the graph in tensorflow directly using tf.layers. .. there is no such problem

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by anitha-v** [comment]
The exp op is now supported , please try it  with nightly.  Please let us know once this works.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by alessiabertugli** [comment]
Has anyone solved this problem? I have the same error with a custom model converted from keras to tflite (.hdf5 to .tflite format). I tested my model in different ways and it seems to work, but when I try to load it in Android It doesn't. Thank you!

**Comment by Cryptiex** [comment]
I have the same problem when converting from .hd5 to .tflite

**Comment by berkgulay** [comment]
Same problem, pre-trained mobilenet architecture on Keras gives this error on Android.

**Comment by GustavZ** [comment]
My Model is not keras but pure tensorflow and still gives this error on android.
But only for the quantized version.

**Comment by julian59189** [comment]
I think it should be:
`interpreter->typed_input_tensor<float>(0)`
instead of
`interpreter->typed_input_tensor<float>(input)`
